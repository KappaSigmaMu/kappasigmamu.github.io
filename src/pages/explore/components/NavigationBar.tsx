import { Nav } from 'react-bootstrap'
import { NavLink } from 'react-router-dom'
import styled from 'styled-components'

const NavigationBar = () => (
  <StyledNav defaultActiveKey="/explore/bidders" className="py-2 my-4">
    <StyledNavItem>
      <Nav.Link as={NavLink} to="/explore/bidders">Bidders</Nav.Link>
    </StyledNavItem>
    <StyledNavItem>
      <Nav.Link as={NavLink} to="/explore/candidates">Candidates</Nav.Link>
    </StyledNavItem>
    <StyledNavItem>
      <Nav.Link as={NavLink} to="/explore/members">Members</Nav.Link>
    </StyledNavItem>
    <StyledNavItem>
      <Nav.Link as={NavLink} to="/explore/suspended">Suspended</Nav.Link>
    </StyledNavItem>
  </StyledNav>
)

const StyledNav = styled(Nav)`
  border-bottom: 1px solid ${(props) => props.theme.colors.darkGrey};
  border-top: 1px solid ${(props) => props.theme.colors.darkGrey};
`

const StyledNavItem = styled(Nav.Item)`
  .nav-link {
    color: ${(props) => props.theme.colors.lightGrey};
  }

  .nav-link:hover {
    color: ${(props) => props.theme.colors.darkGrey};
  }

  .nav-link.active {
    color: ${(props) => props.theme.colors.white};
  }
`

export { NavigationBar }
