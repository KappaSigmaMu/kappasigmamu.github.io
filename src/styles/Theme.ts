import { DefaultTheme } from 'styled-components'

const Theme: DefaultTheme = {
  darkBg: '#212529',
  lightBg: '#ffffff',

  colors: {
    primary: '#E6007A',
    secondary: '#01FFFF',
    white: '#FFFFFF',
    black: '#212529',
    grey: '#ced4da',
    lightGrey: '#6c757d',
    darkGrey: '#343A40',
  },
}

export { Theme }
