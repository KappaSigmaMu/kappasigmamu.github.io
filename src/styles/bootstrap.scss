$theme-colors: (
  'primary': #e6007a,
  'secondary': #01ffff,
  // 'success': red,
  // 'info': red,
  // 'warning': red,
  // 'danger': rgb(204, 193, 193),
  'light': white,
  'dark': black,
  'grey-dark': #6c757d,
);

@import '~bootstrap/scss/bootstrap.scss';
