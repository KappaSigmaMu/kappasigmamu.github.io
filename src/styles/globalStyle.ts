import { createGlobalStyle } from 'styled-components'

const GlobalStyle = createGlobalStyle`
  * {
    color: white;
  }

  html {
    background-color: #212529;
    font-family: Roboto, sans-serif !important;
    font-style: normal;
  }

  body {
    background-color: #212529;
    overflow: auto;
  }

  h1 {
    font-size: 36px;
    font-weight: bold;
  }

  h4 {
    font-size: 16px;
    font-weight: bold;
  }

  h2, h3, h4, h5 {
    color: white;
  }

  a {
    color: #01ffff;
    text-decoration: none;
  }

  a:hover {
    color: white;
  }

  .navbar-light .navbar-nav .nav-link {
    color: #01ffff;
  }

  p {
    color: #c4c4c4;
  }
`

export { GlobalStyle }
