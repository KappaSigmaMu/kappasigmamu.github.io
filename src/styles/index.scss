// Override default variables before the import
$theme-colors: (
  'primary': #e6007a,
  'secondary': #01ffff,
  'dark': black,
  'grey-dark': #6c757d,
  'light': white,
);

* {
  color: white;
}

.dropdown-toggle::after {
  vertical-align: 36% !important;
  margin-left: 0.7em !important;
}

html * {
  font-family: Roboto, sans-serif !important;
  font-style: normal !important;
}

h1 {
  font-weight: bold !important;
  font-size: 36px !important;
}

h4 {
  font-weight: bold !important;
  font-size: 16px !important;
}

h2, h3, h4, h5 {
  color: white;
}

p {
  color: #C4C4C4;
}

a {
  color: #01ffff !important;
}

.btn-lg {
  font-weight: bold !important;
}

.btn-link {
  color: #01ffff !important;
}

// Import Bootstrap and its default variables
@import '~bootstrap/scss/bootstrap.scss';
