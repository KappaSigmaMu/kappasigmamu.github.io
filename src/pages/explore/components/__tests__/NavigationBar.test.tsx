import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { Theme } from '../../../../styles/Theme'
import { NavigationBar } from '../NavigationBar'

describe('NavigationBar', () => {
  it('initialize with explore/bidders selected', () => {
    const tree = renderer.create(
      <MemoryRouter initialEntries={["/explore/bidders"]}>
        <ThemeProvider theme={Theme}>
          <NavigationBar />
        </ThemeProvider>
      </MemoryRouter>
    )

    expect(tree).toMatchSnapshot()
  })
})
