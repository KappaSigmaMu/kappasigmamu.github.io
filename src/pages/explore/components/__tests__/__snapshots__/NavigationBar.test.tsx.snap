// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar initialize with explore/bidders selected 1`] = `
<div
  className="sc-bczRLJ eaeLWo py-2 my-4 nav"
  onKeyDown={[Function]}
>
  <div
    className="sc-gsnTZi hCWTYr nav-item"
  >
    <a
      aria-current="page"
      className="nav-link active"
      data-rr-ui-event-key={null}
      disabled={false}
      href="/explore/bidders"
      onClick={[Function]}
    >
      Bidders
    </a>
  </div>
  <div
    className="sc-gsnTZi hCWTYr nav-item"
  >
    <a
      className="nav-link"
      data-rr-ui-event-key={null}
      disabled={false}
      href="/explore/candidates"
      onClick={[Function]}
    >
      Candidates
    </a>
  </div>
  <div
    className="sc-gsnTZi hCWTYr nav-item"
  >
    <a
      className="nav-link"
      data-rr-ui-event-key={null}
      disabled={false}
      href="/explore/members"
      onClick={[Function]}
    >
      Members
    </a>
  </div>
  <div
    className="sc-gsnTZi hCWTYr nav-item"
  >
    <a
      className="nav-link"
      data-rr-ui-event-key={null}
      disabled={false}
      href="/explore/suspended"
      onClick={[Function]}
    >
      Suspended
    </a>
  </div>
</div>
`;
